src/include/sof/spinlock.h

```diff
@@ -184,36 +184,36 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 
 /* does nothing on UP systems */
 #define spin_lock(lock) \
-	do { \
+	({ \
 		spin_lock_dbg(); \
 		arch_spin_lock(lock); \
-	} while (0)
+	})
 
 #define spin_try_lock(lock, ret) \
-	do { \
+	({ \
 		spin_lock_dbg(); \
 		ret = arch_try_lock(lock); \
-	} while (0)
```

Comment:
This is the only function where this change would have benefitted, and you still used an out parameter instead of actually returning it.